types/node/index.d.ts

```diff
@@ -2017,7 +2004,7 @@ declare module "url" {
         hash?: string;
     }
 
-    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
+    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): URL;
```

Comment:
I have to agree with @Flarna; url.parse() is only available with the legacy API and returns an url.Url object.